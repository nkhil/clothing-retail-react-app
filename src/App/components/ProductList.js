import React, { PureComponent, Fragment } from "react"
import ProductItem from "./ProductItem"

class ProductList extends PureComponent {
  render = () => {
    return (
      <Fragment>
        <ul className="products-list">
          {Object.keys(this.props.products).map(key => (
            <ProductItem
              key={key}
              details={this.props.products[key]}
              index={key}
              addToCart={this.props.addToCart}
              modifyProductInventory={this.props.modifyProductInventory}
            />
          ))}
        </ul>
      </Fragment>
    )
  }
}

export default ProductList
