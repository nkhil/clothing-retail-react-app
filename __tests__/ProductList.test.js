import "@babel/polyfill"
import React from "react"
import ProductList from "../src/App/components/ProductList"
import renderer from "react-test-renderer"

require("react-router-dom")
jest.mock("react-router-dom")

const mockProducts = {
  product1: {
    productName: "Almond Toe Court Shoes, Patent Black",
    productCategory: "Women's Footwear",
    productPrice: 9900,
    productQuantity: 5,
  },
  product2: {
    productName: "Suede Shoes, Blue",
    productCategory: "Women's Footwear",
    productPrice: 4200,
    productQuantity: 4,
  },
}

const mockAddToCart = () => {}
const mockDeductProductFromInventory = () => {}

test("ProductList renders correctly", () => {
  const productList = renderer
    .create(
      <ProductList
        products={mockProducts}
        addToCart={mockAddToCart}
        deductProductFromInventory={mockDeductProductFromInventory}
      />
    )
    .toJSON()
  expect(productList).toMatchSnapshot()
})

// products={this.state.products}
// addToCart={this.addToCart}
// deductProductFromInventory={this.deductProductFromInventory}
