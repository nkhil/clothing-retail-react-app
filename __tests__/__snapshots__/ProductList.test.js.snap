// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<ul>
  <li
    id="product1"
  >
    <h3>
      Almond Toe Court Shoes, Patent Black
    </h3>
    <h4>
      £99.00
    </h4>
    <p>
      Women's Footwear
    </p>
    <button
      className="product1-button"
      onClick={[Function]}
    >
      Add to cart
    </button>
  </li>
  <li
    id="product2"
  >
    <h3>
      Suede Shoes, Blue
    </h3>
    <h4>
      £42.00
    </h4>
    <p>
      Women's Footwear
    </p>
    <button
      className="product2-button"
      onClick={[Function]}
    >
      Add to cart
    </button>
  </li>
</ul>
`;
